suit_generator/cmd_cache_create.py

```diff
@@ -6,116 +6,253 @@
+        """
+        Process list of input binaries, each associated with a URI, and fill the SUIT DFU cache with the data
+
+        :param cache: CachePartition object to fill with the data
+        :param input: List of input binaries with corresponding URIs, passed in the format <URI>,<INPUT_FILE_PATH>
+        """
+
+        for single_input in input:
+            args = single_input.split(",")
+            if len(args) < 2:
+                raise ValueError("Invalid number of input arguments: " + single_input)
+            uri, input_file = args
+
+            with open(input_file, "rb") as f:
+                data = f.read()
+
+            cache.add_cache_slot(uri, data)
+
+class CacheFromEnvelope:
+    def fill_cache_from_envelope_data(cache: CachePartition,
+                                      envelope_data: bytes,
+                                      omit_payload_regex: str,
+                                      dependency_regex: str) -> bytes:
+        """
+        Fill the cache partition with data from the payloads inside the provided envelope binary data.
+        This function is called recursively for dependency envelopes.
 
-        cache_data += slot_data
+        :param cache: CachePartition object to fill with the data
+        :param envelope_data: Binary data of the envelope to extract the payloads from
+        :param omit_payload_regex: Integrated payloads matching the regular expression will not be extracted to the cache
+        :param dependency_regex: Integrated payloads matching the regular expression will be treated as dependency envelopes
+        """
+        envelope = cbor2.loads(envelope_data)
 
-    cache_data += bytes([0xFF])  # Finish the indefinite length map
+        if isinstance(envelope.value, dict):
+            integrated = [k for k in envelope.value.keys() if isinstance(k, str)]
+        else:
+            raise GeneratorError("The provided envelope/dependency envelope is not a valid envelope!")
+
+        if not dependency_regex is None:
+            integrated_dependencies = [k for k in integrated if not re.fullmatch(dependency_regex, k) is None]
+            for dep in integrated_dependencies:
+                integrated.remove(dep)
+
+        if omit_payload_regex is None:
+            payloads_to_extract = integrated
+        else:
+            payloads_to_extract = [k for k in integrated if re.fullmatch(omit_payload_regex, k) is None]
+
+        for payload in payloads_to_extract:
+            cache.add_cache_slot(payload, envelope.value.pop(payload))
+
+        for dependency in integrated_dependencies:
+            new_dependency_data = CacheFromEnvelope.fill_cache_from_envelope_data(
+                cache, envelope.value[dependency], omit_payload_regex, dependency_regex)
+            envelope.value[dependency] = new_dependency_data
+
+        return cbor2.dumps(envelope)
+
+    def fill_cache_from_envelope(cache: CachePartition,
```

Comment:
Although those functions sound similar, the latter writes something to the disk.
Please add some docs about the returned values and side effects as the cache modification is not the only responsibility of these functions.

Reply:
Added a sentence in the function description mentioning that